test/distributed/fsdp/test_fsdp_sharded_grad_scaler.py

```diff
@@ -196,6 +203,168 @@ def test_fsdp_ddp_parity_with_grad_scaler(
                 sharded_grad_scaler_kwargs=sharded_grad_scaler_kwargs,
             )
 
+    def _build_model_and_optim(
+        self,
+        lin_dim: int = 4,
+        build_extra_ddp: bool = False,
+        cpu_offload: CPUOffload = CPUOffload(offload_params=False),
+        use_orig_params: bool = False,
+    ):
+        class MLP(nn.Module):
```

Comment:
It could be worth reusing an existing model in `common_fsdp.py`, e.g. `TransformerWithSharedParams`. You can look at the other FSDP unit tests like for example usage:
https://github.com/pytorch/pytorch/blob/8a8d0adc0b500d70016b71704659dd2d7898e856/test/distributed/fsdp/test_fsdp_use_orig_params.py#L99-L160
This is mainly to avoid creating many per-unit-test toy models when not necessary.